openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -597,6 +618,256 @@ Location: https://client.example.net/cb?
+  }
+}
+```
+
+The Wallet MUST process the Authorization Request contained in the `openid4vp_presentation` parameter as defined in [@!OpenID4VP] to perform a Credential Presentation to the Authorization Server.
+
+For the requested Presentation, the Issuer is acting as a Verifier to the Wallet.
+The exact architecture and the deployment of the Issuer's OpenID4VP Verifier is out of scope of this specification.
+
+In the Authorization Request contained in the `openid4vp_presentation` parameter, the `response_type` MUST be `vp_token` (defined in Section 8 of [@!OpenID4VP]).
+
+The `response_mode` of the request MUST be `direct_post` (defined in Section 8.2 of [@!OpenID4VP]) or `direct_post.jwt` (defined in Section 8.3.1 of [@!OpenID4VP]).
+Following the definition of this response mode, the Wallet sends an HTTP POST request to the endpoint indicated by the `response_uri` parameter in the Authorization Request to either complete the presentation (sending the VP Token) or to indicate an error.
+The Verifier MUST respond to this message by sending a JSON object containing a `redirect_uri` parameter.
+
+Note: Sending `redirect_uri` is defined as OPTIONAL in [@!OpenID4VP], but it is REQUIRED here.
+
+In a regular presentation flow, the Wallet would be expected to follow this redirect.
+In the case described here, the Wallet MUST NOT follow the redirect URI and MUST instead repeat the request to the Interactive Authorization Endpoint and in this request include the received redirect URI in the `interactive_binding_token` parameter.
+The Wallet MUST NOT modify the URI in any way and treat it as an opaque value.
+The Issuer MUST verify that the redirect URI in the `interactive_binding_token` parameter is correct, i.e., matches the one sent in response to the request to the `response_uri`.
+Since the redirect URI MUST include a fresh, cryptographically random value, this check helps to present Session Fixation attacks, see (#iar-security).
+
+#### Redirect to Web
+
+If the type is `redirect_to_web`, the Authorization Server wants to fall back to a regular interaction with the user.
+
+In this case, the Authorization server MUST include the key `request_uri` in the response.
+The Wallet MUST use the `request_uri` value to build an Authorization Request as defined in Section 4 of [@!RFC9126] and complete the rest of the authorization process as defined there.
+The Authorization Server MAY include the `expires_in` key as defined in [@!RFC9126].
+
+Non-normative Example:
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "require_interaction",
+  "type": "redirect_to_web",
+  "request_uri": "urn:ietf:params:oauth:request_uri:6esc_11ACC5bwc014ltc14eY22c",
+  "expires_in": 60
+}
+```
+
+#### Custom Interaction Extensions {#iar-custom-extensions}
+
+Additional, custom types of interactions MAY be defined by extensions of this specification to enable other types of interactions, for example, scanning of an NFC card.
```

Comment:
Not entirely sure right now how to better word this, but if I recall correctly we wanted to explicitly mention smart cards here (fitting to the example below)? scanning an NFC card sounds kind of wrong to me?

Not sure this is better, maybe @paulbastian has a better wording?
```suggestion
Additional, custom types of interactions MAY be defined by extensions of this specification to enable other types of interactions, for example, by interacting with a smart card.
```